src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1159,13 +1171,12 @@ def _configure_security_policy_storage_params(arg_ctx):
                    id_part='name')
 
         c.argument('tier',
-                   arg_type=tier_param_type,
-                   help='The edition component of the sku. Allowed values: GeneralPurpose, BusinessCritical.')
+                   arg_type=mi_tier_param_type,
+                   help='The edition component of the sku.')
 
         c.argument('family',
-                   arg_type=family_param_type,
-                   help='The compute generation component of the sku. '
-                   'Allowed values include: Gen4, Gen5.')
+                   arg_type=mi_family_param_type,
+                   help='The compute generation component of the sku.')
```

Comment:
Does `get_enum_type()` throw error if you don't specify one of the allowed values? This will prevent forward compatibility (i.e. when we release Gen1234, customers will not be able to manually type it in until they download newer CLI version) - this has previously been a frustration for SQL DB customers. We should allow customer to type in anything, then if we validate it must be against a dynamic list (e.g. capabilities API) and NOT a hardcoded list.

Reply:
I've checked now and it does throws an error. For example

az sql mi update --subscription "DS-CloudLifter_jovanc_R&D_60843" --resource-group 201tmpl --name nestan-testing109 --family Gen52
az sql mi update: 'Gen52' is not a valid value for '--family'. See 'az sql mi update --help'.

The most similar choice to 'Gen52' is:
        Gen5

I'll remove the get_enum_type :)

Reply:
Removed

Reply:
Thanks for pointing this up